src/_gettsim/transfers/erwerbsm_rente.py

```diff
@@ -0,0 +1,358 @@
+
+    Returns
+    -------
+    average entgeltp
+    """
+
+    beleg_gesamtzeitr = (
+        age_of_retirement - erwerbsm_rente_params["altersgrenze_grundbew"]
+    )
+
+    entgeltp_gesamt = entgeltp_west + entgeltp_ost
+
+    durchschn_entgeltp = entgeltp_gesamt / beleg_gesamtzeitr
+
+    return durchschn_entgeltp
+
+
+def rentenartfaktor(
+    teilw_erwerbsm_rente: bool,
+    erwerbsm_rente_params: dict,
+) -> float:
+    """Rentenartfaktor for Erwerbsminderungsrente
+    (public disability insurance)
+
+    Legal reference: SGB VI § 67: Rentenartfaktor
+
+    Parameters
+    ----------
+    teilw_erwerbsm_rente
+        See basic input variable :ref:`teilw_erwerbsm_rente <teilw_erwerbsm_rente>.
+    erwerbsm_rente_params
+        See params documentation :ref:`erwerbsm_rente_params <erwerbsm_rente_params>.
+
+    Returns
+    -------
+    Rentenartfaktor
+
+    """
+
+    if teilw_erwerbsm_rente:
+        out = erwerbsm_rente_params["rentenartfaktor"]["teilw"]
+
+    else:
+        out = erwerbsm_rente_params["rentenartfaktor"]["voll"]
+
+    return out
+
+
+def erwerbsm_rente_zugangsfaktor(
+    ges_rente_params: dict,
+    erwerbsm_rente_params: dict,
+    age_of_retirement: float,
+    langj_versicherte_wartezeit: bool,
+) -> float:
+    """Zugangsfaktor for Erwerbsminderungsrente
+    (public disability insurance)
+    For each month that a pensioner retires before the age limit, a fraction of
+    the pension is deducted. The maximum deduction is capped.
+    This max deduction is the norm for the public disability insurance.
+
+    Legal reference: § 77 Abs. 2-4  SGB VI
+
+    Paragraph 4 regulates an exceptional case in which pensioners can already
+    retire at 63 without deductions if they can prove 40 years of
+    (Pflichtbeiträge, Berücksichtigungszeiten and
+    certain Anrechnungszeiten or Ersatzzeiten).
+
+    Parameters
+    ----------
+    ges_rente_params
+        See params documentation :ref:`ges_rente_params <ges_rente_params>.
+    erwerbsm_rente_params
+        See params documentation :ref:`erwerbsm_rente_params <erwerbsm_rente_params>.
+    age_of_retirement
+        See :func:`age_of_retirement`.
+    langj_versicherte_wartezeit
+        See :func:`langj_versicherte_wartezeit`.
+
+
+    Returns
+    -------
+    Zugangsfaktor for Erwerbsminderungsrente (public disability insurance)
+
+    """
+
+    if langj_versicherte_wartezeit:
+        altersgrenze_abschlagsfrei = erwerbsm_rente_params[
+            "altersgrenze_langj_versicherte_abschlagsfrei"
+        ]
+    else:
+        altersgrenze_abschlagsfrei = erwerbsm_rente_params["altersgrenze_abschlagsfrei"]
+
+    zugangsfaktor = 1 + (age_of_retirement - altersgrenze_abschlagsfrei) * (
+        ges_rente_params["zugangsfaktor_veränderung_pro_jahr"][
+            "vorzeitiger_renteneintritt"
+        ]
+    )
+    out = max(zugangsfaktor, erwerbsm_rente_params["min_zugangsfaktor"])
+
+    return out
+
+
+def langj_versicherte_wartezeit(  # noqa: PLR0913
```

Comment:
```suggestion
def _erwerbsm_rente_langj_versicherte_wartezeit(  # noqa: PLR0913
```

For now we'll need to be explicit about this.